engine.py

```diff
@@ -190,114 +223,61 @@ def log_error(self, msg):
-                
-                base["qt_core"] = QtCore
-                base["qt_gui"] = QtGui
-                base["dialog_base"] = ProxyDialogPySide
-                self.log_debug("Successfully initialized PySide '%s' located in %s." 
-                               % (PySide.__version__, PySide.__file__))
-                self._has_qt = True
-            except ImportError:
-                pass
-            except Exception, e:
-                self.log_warning("Error setting up pyside. Pyside based UI support will not "
-                                 "be available: %s" % e)
-        
-        if not self._has_qt:
-            try:
-                from PyQt4 import QtCore, QtGui
-                import PyQt4
-                
-                # tell QT to interpret C strings as utf-8
-                utf8 = QtCore.QTextCodec.codecForName("utf-8")
-                QtCore.QTextCodec.setCodecForCStrings(utf8)                
-                
-                # a simple dialog proxy that pushes the window forward
-                class ProxyDialogPyQt(QtGui.QDialog):
-                    def show(self):
-                        QtGui.QDialog.show(self)
-                        self.activateWindow()
-                        self.raise_()
-                
-                    def exec_(self):
-                        self.activateWindow()
-                        self.raise_()
-                        # the trick of activating + raising does not seem to be enough for
-                        # modal dialogs. So force put them on top as well.                        
-                        self.setWindowFlags(QtCore.Qt.WindowStaysOnTopHint | self.windowFlags())
-                        QtGui.QDialog.exec_(self)
-                
-                
-                # hot patch the library to make it work with pyside code
-                QtCore.Signal = QtCore.pyqtSignal     
-                QtCore.Slot = QtCore.pyqtSlot
-                QtCore.Property = QtCore.pyqtProperty           
-                base["qt_core"] = QtCore
-                base["qt_gui"] = QtGui
-                base["dialog_base"] = ProxyDialogPyQt
-                self.log_debug("Successfully initialized PyQt '%s' located in %s." 
-                               % (QtCore.PYQT_VERSION_STR, PyQt4.__file__))
-                self._has_qt = True
-            except ImportError:
-                pass
-            except Exception, e:
-                self.log_warning("Error setting up PyQt. PyQt based UI support will not "
-                                 "be available: %s" % e)
-        
+        # Just call the base implementation and monkey patch QMessageBox.
+        base = super(ShotgunEngine, self)._define_qt_base()
+
+        if not base:
+            self._has_qt = False
+
+            # proxy class used when QT does not exist on the system.
+            # this will raise an exception when any QT code tries to use it
+            class QTProxy(object):
```

Comment:
We should at some point return that from tk-core so we can rid of this boiler plate in tk-shotgun and tk-shell.

Reply:
yes agree!

Reply:
hmm i looked at this and i think it might be tricky to do - i think we have a bunch of code which checks for None values and if we return proxy objects instead it will be confusing.

Reply:
Ah yes you're right, I remember now. This is why we didn't do it last time!